Boccia-Unity/Assets/Boccia/Ramp/BallPresenter.cs

```diff
@@ -6,42 +6,46 @@ public class BallPresenter : MonoBehaviour
 {
     public GameObject ball; // The ball prefab
     private GameObject activeBall; // Refers the the ball currently in use for each shot
-    private int ballCount = 1;
+    private int ballCount = 0;
     private Rigidbody ballRigidbody;
 
     private Animator barAnimation;
     public GameObject dropBar;
-
     public GameObject elevationPlate;
 
     private BocciaModel model;
 
     private Vector3 dropPosition;
     private Quaternion dropRotation;
 
+    private Coroutine dropBallCoroutine;
     private Coroutine checkBallCoroutine;
 
+    private bool firstBallDropped = false; // To check if at least one ball has been dropped
+
     // Start is called before the first frame update
     void Start()
     {
         // cache model and subscribe for changed event
         model = BocciaModel.Instance;
-        model.WasChanged += ModelChanged;
+        model.WasChanged += RampChanged;
+        model.BallResetChanged += ResetBocciaBalls;
```

Comment:
Maybe this should be OnEnabled()? 
Otherwise you will remove this OnDisable and won't ever add it back

Reply:
Good point, but since BallPresenter is attached to the Boccia object that shouldn't be an issue for now (I don't think Boccia is ever disabled at any point). I can add that anyway though.